static/app/components/performance/spanSearchQueryBuilder.tsx

```diff
@@ -0,0 +1,101 @@
+import {useCallback, useMemo} from 'react';
+
+import {fetchSpanFieldValues} from 'sentry/actionCreators/tags';
+import {normalizeDateTimeParams} from 'sentry/components/organizations/pageFilters/parse';
+import {SearchQueryBuilder} from 'sentry/components/searchQueryBuilder';
+import type {CallbackSearchState} from 'sentry/components/searchQueryBuilder/types';
+import {t} from 'sentry/locale';
+import type {PageFilters} from 'sentry/types/core';
+import {SavedSearchType, type Tag, type TagCollection} from 'sentry/types/group';
+import {defined} from 'sentry/utils';
+import {isAggregateField, isMeasurement} from 'sentry/utils/discover/fields';
+import {DEVICE_CLASS_TAG_VALUES, isDeviceClass} from 'sentry/utils/fields';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {useSpanFieldSupportedTags} from 'sentry/views/performance/utils/useSpanFieldSupportedTags';
+
+interface SpanSearchQueryBuilderProps {
+  initialQuery: string;
+  searchSource: string;
+  datetime?: PageFilters['datetime'];
+  disableLoadingTags?: boolean;
+  onSearch?: (query: string, state: CallbackSearchState) => void;
+  placeholder?: string;
+  projects?: PageFilters['projects'];
+}
+
+export function SpanSearchQueryBuilder({
+  initialQuery,
+  searchSource,
+  datetime,
+  onSearch,
+  placeholder,
+  projects,
+}: SpanSearchQueryBuilderProps) {
+  const api = useApi();
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+
+  const placeholderText = useMemo(() => {
```

Comment:
nit: memo not necessary for string props

Reply:
will address in the follow up